src/Routing/Router.php

```diff
@@ -1148,6 +1148,17 @@ public static function getRouteCollection()
         return static::$_collection;
     }
 
+    /**
+     * Set the RouteCollection inside the Router
+     *
+     * @param RouteCollection $routeCollection route collection
+     * @return void
+     */
+    public static function setRouteCollection($routeCollection)
+    {
+        static::$_collection = $routeCollection;
```

Comment:
You can set the `$initialized` property here, and avoid to do it in the middleware

Reply:
that makes a lot of sense, thanks @lorenzo